src/h5web/visualizations/containers/NxSpectrumContainer.tsx

```diff
@@ -16,9 +21,16 @@ function NxSpectrumContainer(props: VisContainerProps): ReactElement {
   assertGroup(entity);
 
   const { metadata } = useContext(ProviderContext);
-  const signal = getAttributeValue(entity, 'signal');
+
+  const NxData = getNxDataGroup(entity, metadata);
+  if (!NxData) {
+    throw new Error('NXdata group not found');
+  }
+  assertGroup(NxData);
```

Comment:
This assertion is not needed since the function is already typed to return a group.